CaravelLabs/sunrise-2024
# Implemented task management features as guided

Sunrise 2024 Taskboard System Implementation
============================================

Overview
--------

This PR implements the Taskboard System designed to streamline the process of assigning, managing, and tracking tasks for employees within an organization. The system ensures that tasks are completed sequentially and efficiently, providing a clear progression path for employees as they work through their assignments.

Features Implemented
--------------------

### Task Management

-   **Sequential Task Generation:** Tasks are unlocked sequentially, requiring employees to complete all tasks within a group before the next group becomes available. Upon completion of a task, the next task in the sequence is automatically activated.
-   **Auto-Initialization:** The first task is automatically created and assigned to an employee upon the initialization of the program.
-   **Active Tasks View:** Employees can view their current active tasks.
-   **Task Completion:** Employees can mark tasks as completed.
-   **Task Creation:** Ability to create new tasks through a dedicated API.
-   **Task Update:** Allows updating task details, including status, through an API.
-   **Task Deletion:** Tasks can be deleted from the system if needed.


Technology Stack
----------------
-   **Next.js:** Utilized for building the web application, ensuring server-side rendering and seamless API integration.
-   **Ant Design:** Used for creating a modern and responsive UI.
-   **TypeScript:** Ensured type safety across the application, making the code more robust and easier to maintain.

Screenshots and Video
---------------------

Below are screenshots and a video demonstrating the functionality of the Taskboard System:
-   **Video:**

    -   Video 1: Walkthrough of the Taskboard System, including task creation, updating, and deletion.

https://github.com/user-attachments/assets/3c531029-6296-4712-8584-5fb8d04dc1c5

-   **Screenshots:**

    -   Screenshot 1: Active Tasks View
![1](https://github.com/user-attachments/assets/8006f5c1-9b51-498f-adb4-cf3dffdf4e79)
    -   Screenshot 2: Task Marking Done
![2](https://github.com/user-attachments/assets/b5b4974d-d997-43e7-abe8-0d7661b8ab80)
    -    Screenshot 3: Task Updating
![Screenshot (1492)](https://github.com/user-attachments/assets/b3b3559c-cfac-48c5-8a1d-7e743a08bfce)
    -  Screenshot 4: Task Creation 
    ![4](https://github.com/user-attachments/assets/d5489531-4b0e-4d8e-9cfd-e2caee36e7b8)
    -   Screenshot 5: After Creating
     ![5](https://github.com/user-attachments/assets/7311c69f-6675-4b92-b360-69540014e4e1)
    -  Screenshot 6: After all Task Completed
   ![3](https://github.com/user-attachments/assets/4808a992-a501-4689-9a3d-166c516d44b9)
 




How to Test
-----------

1.  Clone the repository and navigate to the `web` folder.
2.  Install the dependencies using `npm install`.
3.  Start the development server with `npm run dev`.
4.  Access the application at `http://localhost:3000` and interact with the Taskboard System.
5.  Run tests with `npm run test` to ensure all functionalities work as expected.
